crates/dslab-dag/src/tests.rs

```diff
@@ -220,30 +224,67 @@ fn test_4() {
     add_edge(7, 9, 11, "n");
     add_edge(8, 9, 13, "o");
 
-    let mut sim = DagSimulation::new(
-        123,
-        Rc::new(RefCell::new(ConstantBandwidthNetwork::new(1.0, 0.0))),
-        Rc::new(RefCell::new(HeftScheduler::new())),
-        Config {
-            data_transfer_mode: DataTransferMode::Direct,
-        },
-    );
-    sim.add_resource("0", 1, 1, 0);
-    sim.add_resource("1", 2, 1, 0);
-    sim.add_resource("2", 4, 1, 0);
-    sim.add_resource("3", 4, 1, 0);
-
-    let runner = sim.init(dag);
-    sim.step_until_no_events();
-    assert!(runner.borrow().is_completed());
+    fn run_scheduler(scheduler: impl Scheduler + 'static, dag: DAG) -> f64 {
+        let mut sim = DagSimulation::new(
+            123,
+            Rc::new(RefCell::new(ConstantBandwidthNetwork::new(1.0, 0.0))),
+            Rc::new(RefCell::new(scheduler)),
+            Config {
+                data_transfer_mode: DataTransferMode::Direct,
+            },
+        );
+        sim.add_resource("0", 1, 1, 0);
+        sim.add_resource("1", 2, 1, 0);
+        sim.add_resource("2", 4, 1, 0);
+        sim.add_resource("3", 4, 1, 0);
+
+        let runner = sim.init(dag);
+        runner.borrow_mut().enable_trace_log(true);
+        sim.step_until_no_events();
+        assert!(runner.borrow().is_completed());
+
+        runner.borrow().trace_log().save_to_file("simple.log").unwrap();
+        return sim.time();
+    }
 
-    let result = sim.time();
-    assert_eq!(result, 98.0);
+    let heft_makespan = run_scheduler(HeftScheduler::new(), dag.clone());
 
     // 0:
     // 1:                           [-------------E------------]
     // 2:[-------A------][-----C-----][--------B--------][------G------]
     // 3:                              [-------F------][---D--]              [-----I----][----H----][--J--]
+
+    assert_eq!(heft_makespan, 98.0);
+
+    let lookahead_makespan = run_scheduler(LookaheadScheduler::new(), dag.clone());
+
+    // 0:
+    // 1:                         [-------D------]
+    // 2:[-------A------][-----C-----][--------B--------][-------F------][-----I----][----H----][--J--]
+    // 3:                           [------E-----]           [------G------]
+
+    assert_eq!(lookahead_makespan, 94.0);
```

Comment:
Здесь и у других алгоритмов пустые строки вокруг расписания не нужны:

```suggestion
    let lookahead_makespan = run_scheduler(LookaheadScheduler::new(), dag.clone());

    // 0:
    // 1:                         [-------D------]
    // 2:[-------A------][-----C-----][--------B--------][-------F------][-----I----][----H----][--J--]
    // 3:                           [------E-----]           [------G------]

    assert_eq!(lookahead_makespan, 94.0);
```

Reply:
Осталось убрать пустые строки и можно мержить